src/main/java/com/points/node/binarytree/BinaryTreeBuilder.java

```diff
@@ -0,0 +1,58 @@
+package com.points.node.binarytree;
+
+import java.util.List;
+
+import com.points.node.twochild.Node;
+
+public class BinaryTreeBuilder {
+
+	public static <T extends Comparable<T>> Node<T> setUpTree(List<T> nodeValues) {
+		if (nodeValues == null) {
+			return null;
+		}
+
+		Node<T> firstNode = null;
+		for (T value : nodeValues) {
+			if (firstNode == null) {
+				firstNode = new Node<>(value);
+				continue;
```

Comment:
You could simplify this by setting firstNode to the first node in the list. Then iterating the list starting with the second element.